js/ui/html_editor/modules/toolbar.js

```diff
@@ -503,20 +503,29 @@ if(Quill) {
                     continue;
                 }
 
-                this._markActiveFormatWidget(formatName, formatWidget, formats);
+                this._markActiveFormatWidget(formatName, formatWidget, formats, selection);
             }
 
             this._toggleClearFormatting(hasFormats || selection.length > 1);
         }
 
-        _markActiveFormatWidget(name, widget, formats) {
+        _markActiveFormatWidget(name, widget, formats, selection) {
             if(this._isColorFormat(name)) {
                 this._updateColorWidget(name, formats[name]);
             }
 
             if('value' in widget.option()) {
                 this._setValueSilent(widget, formats[name]);
             } else {
+                if(selection?.length === 0) { // T1157840
+                    const nextPositionFormat = this.quill.getFormat({
```

Comment:
It's more concrete and native for the HtmlEditor
```suggestion
                    const nextIndexFormat = this.quill.getFormat({
```